File: LeetCode/Easy/1768-merge-strings-alternately/1768-merge-strings-alternately.cpp
class Solution {
public:
    string mergeAlternately(string word1, string word2) {
        int idx1 = 0, idx2 = 0;
        string res = "";
        
        while (idx1 < word1.length() && idx2 < word2.length()) {
            res += word1.at(idx1++);
            res += word2.at(idx2++);
        }
        
        while (idx1 < word1.length()) {
            res += word1.at(idx1++);
        }
        while (idx2 < word2.length()) {
            res += word2.at(idx2++);
        }
        return res;
    }
};

File: LeetCode/Easy/1768-merge-strings-alternately/1768-merge-strings-alternately.js
/**
 * @param {string} word1
 * @param {string} word2
 * @return {string}
 */
var mergeAlternately = function(word1, word2) {
    var idx = 0;
    var res = "";
    
    while (idx < word1.length && idx < word2.length) {
        res += word1.charAt(idx);
        res += word2.charAt(idx);
        idx++;
    }
    while(idx < word1.length) {
        res += word1.charAt(idx++);
    }
    while(idx < word2.length) {
        res += word2.charAt(idx++);
    }
    return res
};